atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -274,28 +274,33 @@ def run(self):
                                                             user_provided_release)
 
             if self.reserve_build:
-                self.reserve_build_in_koji(component, version, user_provided_release,
-                                           release_label, dockerfile_labels,
-                                           user_provided_release=True)
+                self.reserve_build_in_koji(
+                    component, version, user_provided_release, explicitly_provided_release=True
+                )
             return
 
-        if release and not self.append:
+        if original_release and not self.append:
             self.log.debug("release set explicitly so not incrementing")
+            release = original_release
 
             if not is_scratch_build(self.workflow):
                 self.check_build_existence_for_explicit_release(component, version, release)
-                dockerfile_labels[release_label] = release
         else:
             # release not set or release should be appended
-            next_release = self.next_release_general(component, version, release)
-            dockerfile_labels[release_label] = next_release
+            release = self.next_release_general(component, version, original_release)
 
         if self.reserve_build and not is_scratch_build(self.workflow):
-            self.reserve_build_in_koji(
-                component, version, release, release_label, dockerfile_labels
+            release = self.reserve_build_in_koji(
+                component, version, release, explicitly_provided_release=bool(original_release)
             )
 
-        # TODO: add the log message about setting release label
+        # Always set preferred release label - other will be set if old-style label is present
+        preferred_release_label = Labels.LABEL_NAMES[Labels.LABEL_TYPE_RELEASE][0]
+
+        if dockerfile_labels.get(preferred_release_label) != release:
+            self.log.info("setting %s=%s", preferred_release_label, release)
+            # Write the label back to the file (this is a property setter)
+            dockerfile_labels[preferred_release_label] = release
```

Comment:
Unclear: by removing the "redundant" early return in a previous commit, this is now a change in behavior for scratch builds that have a release label in the Dockerfile. Previously, we wouldn't set the "preferred" release label for those, now we will.

Is there any reason why we wouldn't want to set the preferred release label for scratch builds?

Reply:
no, it just wasn't necessary since for scratch builds bumping release is just formality not really requirement

Reply:
Ack, I'll keep it like this then